source/javascripts/controllers/_StackController.js.erb

```diff
@@ -141,10 +159,12 @@
 			viewModel.onDefaultStackChange = function() {
 				appService.dockerImage = viewModel.defaultDockerImagesByStackIDs[appService.stack.id] ? '' : null;
 
-				var isPreviousMachineTypeAvailable = viewModel.isMachineTypeAvailableForStack(appService.defaultMachineType, appService.stack);
+				if(!appService.stack.id.startsWith('agent-pool')) {
+					var isPreviousMachineTypeAvailable = viewModel.isMachineTypeAvailableForStack(appService.defaultMachineType, appService.stack);
 
-				if (!isPreviousMachineTypeAvailable) {
-					viewModel.setDefaultMachineType(viewModel.firstAvailableMachineTypeForStack(appService.stack.id));
+					if (!isPreviousMachineTypeAvailable) {
+						viewModel.setDefaultMachineType(viewModel.firstAvailableMachineTypeForStack(appService.stack.id));
+					}
```

Comment:
In `Workflow.stack` instance method we explicitly delete the machineType if the stack is an agent pool

```
if(stack.isAgentPoolStack()) {
  delete this.workflowConfig.meta[BITRISE_META_KEY].machine_type_id;
}
```

This way it will never be set in the yml output. I thought that this solves the problem without introducing a case where `defaultMachineType` will be null.

Reply:
That `Workflow.stack` method only does workflow-specific changes, it won't alter the default stack/machine.